### PR TITLE
add h2o-py

### PR DESCRIPTION
**Pull request by fhoehle**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by fhoehle** [comment]
@conda-forge/help-python This is ready for review.

File: recipes/h2o-py/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  name: "{{ name|lower }}-py"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: "839291a0f74b18790120ecc16343cede545dac668a953fc94455775022104dda"
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
```
Comment (scopatz) [inline recipes/h2o-py/meta.yaml]: can this be noarch: python?

Comment (fhoehle) [reply recipes/h2o-py/meta.yaml]: @scopatz thanks for spotting this! fixed.

**Comment by xhochy** [comment]
Three things come to my mind here:

* What about basing this on https://github.com/AnacondaRecipes/h2o-feedstock/blob/master/recipe/meta.yaml
* What about the naming? In defaults it is named `h2o-py` whereas in the `h20ai` conda channel the Python package is called `h2o`.
* As upstream provides a conda-package, maybe make them aware of the PR.

**Comment by xhochy** [comment]
Upstream h2o recipe: https://github.com/h2oai/h2o-3/blob/master/h2o-py/conda/h2o/meta.yaml

**Comment by fhoehle** [comment]
The anaconda team tried to provide h2o as `jar`, `python` and `R` package. The R build step was not functional. The h2oai team provides a conda recipe only for python. I would propose to be flexible and have room for several h2o packages corresponding to the different languages to name the python package `h2o-py` and, as initial step, to mirror the h2oai conda recipe.

**Pull request closed without merging**